File: docs/chuanglanDoc/PYTHON/sms.py
#!/usr/local/bin/python
#-*- coding:utf-8 -*-
# Author: jacky
# Time: 14-2-22 下午11:48
# Desc: 短信http接口的python代码调用示例
import httplib
import urllib
import json
#服务地址
host = "vsms.253.com"

#端口号
port = 80

#版本号
version = "v1.1"

#查账户信息的URI
balance_get_uri = "/msg/balance/json"

#智能匹配模版短信接口的URI
sms_send_uri = "/msg/send/json"

#创蓝账号
account  = "N9994784"

#创蓝密码
password = "Jiubugaosuni10"

def get_user_balance():
    """
    取账户余额
    """
    params = {'account': account, 'password' : password}
    params=json.dumps(params)
   
    headers = {"Content-type": "application/json"}
    conn = httplib.HTTPConnection(host, port=port)
    conn.request('POST', balance_get_uri, params, headers)
    response = conn.getresponse()
    response_str = response.read()
    conn.close()
    return response_str

def send_sms(text, phone):
    """
    能用接口发短信
    """
 
    params = {'account': account, 'password' : password, 'msg': urllib.quote(text), 'phone':phone, 'report' : 'false'}
    params=json.dumps(params)
   
    headers = {"Content-type": "application/json"}
    conn = httplib.HTTPConnection(host, port=port, timeout=30)
    conn.request("POST", sms_send_uri, params, headers)
    response = conn.getresponse()
    response_str = response.read()
    conn.close()
    return response_str 

if __name__ == '__main__':

    phone = "18721755342"
    text = "【253云通讯】您的验证码是1234"

    #查账户余额
    print(get_user_balance())

    #调用智能匹配模版接口发短信
    print(send_sms(text, phone))

File: docs/chuanglanDoc/RUBY/sms.rb
# 说明：
# 以下代码只是为了方便客户测试而提供的示例代码，客户可以根据自己的需要另行编写
# 该代码仅供学习和研究接口使用，只是提供了一个参考

require 'net/http'  
require 'uri'  
require 'json'  
  
params = {  
    "account" => "",  
    "password" => "a.123456",  
    "phone" => "18721755342",   
    "msg" =>URI::escape("【253云通讯】您好，您的验证码是999999") 
}.to_json  
  
def send_data(url,data)  
    url = URI.parse(url)  
    req = Net::HTTP::Post.new(url.path,{'Content-Type' => 'application/json'})  
    req.body = data  
    res = Net::HTTP.new(url.host,url.port).start{|http| http.request(req)}  
  
    puts res.body                                                                                                  
end  

send_data('http://vsms.253.com/msg/send/json',params) 